[BUG] The `before` date in `.npmrc` is not parsed as `Date` so every install fails

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

As in the title, this `.npmrc` file will cause installs to fail because the value is passed to pacote as a `string` while pacote expects a `Date`:

```
before=2022-10-26
```

I'm opening the issue here because it can be fixed here, but it's possible this is a bug within `npm-cli` instead. I wasn't able to find the exact part that dealt with this conversion elsewhere.

https://github.com/npm/pacote/blob/7a560db966140db65b8f000de4d36eb36e0a7760/lib/fetcher.js#L125

```
❯ npm ci
npm ERR! this.before.toISOString is not a function

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2023-03-09T16_29_48_321Z-debug-0.log
```

### Expected Behavior

A date in .npmrc should be handled the same way as a date passed as `npm ci --before=2022-10-26T00:00:00.000Z`

I don't know if a fix belongs here, but in my case it might be _as easy as_ `typeof before === string ? new Date(before) : before`

### Steps To Reproduce


```sh
npm init -y
echo 'before=2022-10-26T00:00:00.000Z' > .npmrc
npm install ky
```

### Environment

- npm: 9.6.0
- Node: v18.13.0.
- OS: macOS
- platform:

--- a/workspaces/config/lib/parse-field.js
+++ b/workspaces/config/lib/parse-field.js
@@ -20,6 +20,7 @@ const parseField = (f, key, opts, listElement = false) => {
   const isUmask = typeList.has(typeDefs.Umask.type)
   const isNumber = typeList.has(typeDefs.Number.type)
   const isList = !listElement && typeList.has(Array)
+  const isDate = typeList.has(typeDefs.Date.type)
 
   if (Array.isArray(f)) {
     return !isList ? f : f.map(field => parseField(field, key, opts, true))
@@ -52,6 +53,10 @@ const parseField = (f, key, opts, listElement = false) => {
   }
 
   f = envReplace(f, env)
+
+  if (isDate) {
+    return new Date(f)
+  }
 
   if (isPath) {
     const homePattern = platform === 'win32' ? /^~(\/|\\)/ : /^~\//

--- a/workspaces/config/test/parse-field.js
+++ b/workspaces/config/test/parse-field.js
@@ -34,3 +34,5 @@ t.equal(parseField('1234', 'maxsockets', opts), 1234, 'number is parsed')
 t.equal(parseField('0888', 'umask', opts), '0888',
   'invalid umask is not parsed (will warn later)')
 t.equal(parseField('0777', 'umask', opts), 0o777, 'valid umask is parsed')
+
+t.same(parseField('2020', 'before', opts), new Date('2020'), 'date is parsed')

